qiskit/pulse/commands/acquire.py

```diff
@@ -22,14 +22,18 @@
 from .instruction import Instruction
 from .meas_opts import Discriminator, Kernel
 from .command import Command
+import itertools
 
 
 class Acquire(Command):
     """Acquire."""
 
     ALIAS = 'acquire'
 
-    def __init__(self, duration, discriminator=None, kernel=None):
+    # Counter for the number of instances in this class
+    instances_counter = itertools.count(0)
```

Comment:
This pattern of initiating a new counter for each subclass could be resolved with a Metaclass, for [example see] (https://stackoverflow.com/a/44679581).

Reply:
@taalexander I had a stab at using a Metaclass, I believe the implementation works (at least according to the tests, I also added a test to test the incrementing of the class counter). However, `make style && make lint` are failing due to lines 62 and 82 in command.py (error is `'Command' has no 'instances_counter' member`) but tests show that the counter is incrementing and the code does not crash. Any ideas what is happening here?

Reply:
I think this is a case of pylint not understanding what the metaclass is doing.